src/JuliusSweetland.OptiKey/UI/ViewModels/MainViewModel.cs

```diff
@@ -127,6 +127,21 @@ public IKeyboard Keyboard
             }
         }
 
+        public void BackFromKeyboard()
+        {
+            Log.Info("Navigating back from keyboard via contextmenu.");
+            var navigableKeyboard = Keyboard as IBackAction;
+            if (navigableKeyboard != null && navigableKeyboard.BackAction != null)
+            {
+                navigableKeyboard.BackAction();
+            }
+            else
+            {
+                // TODO: Shouldn't we always remember 'last' keyboard?
+                Log.Error("Keyboard doesn't have backaction, can't go back");
```

Comment:
Instead this should mirror the "case FunctionKeys.BackFromKeyboard:" handler and go to the default/startup keyboard.